From c6b0faef4a901dc14a016520a3f9cfa5f763b78b Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Tue, 2 Feb 2021 00:11:30 +0000
Subject: [PATCH] Draft radiance

Build fails with r/w permission to ubiquitous location.
For someone who is brave do dig deeper:

phase `patch-generated-file-shebangs' succeeded after 0.2 seconds
starting phase `build'
Invoking sbcl: "/gnu/store/wyfbw52srslpf3cbyjzfxhd16n7ilk91-sbcl-2.1.0/bin/sbcl" "--non-interactive" "--eval" "(require :asdf)" "--eval" "(asdf:load-asd (truename \"/gnu/store/301jp24ywjpmaax0vx313fwgg0p9x0ds-sbcl-radiance-2.1.2-1.5ffbe1f/share/common-lisp/sbcl/radiance/rad
iance-core.asd\"))" "--eval" "(asdf:load-asd (truename \"/gnu/store/301jp24ywjpmaax0vx313fwgg0p9x0ds-sbcl-radiance-2.1.2-1.5ffbe1f/share/common-lisp/sbcl/radiance/radiance.asd\"))" "--eval" "(asdf:compile-system \"radiance-core\")" "--eval" "(asdf:compile-system \"radiance\
")"
This is SBCL 2.1.0, an implementation of ANSI Common Lisp.
More information about SBCL is available at <http://www.sbcl.org/>.

SBCL is free software, provided as is, with absolutely no warranty.
It is mostly in the public domain; some portions are provided under
BSD-style licenses.  See the CREDITS and COPYING files in the
distribution for more information.
Unhandled SB-INT:SIMPLE-FILE-ERROR in thread #<SB-THREAD:THREAD "main thread" RUNNING
                                                {1001878103}>:
  Error opening #P"/gnu/store/c4fj8gfdshgbfrm1jq3rdg55m24mkjh3-sbcl-ubiquitous-2.0.0-1.35eb7bd/lib/common-lisp/sbcl/ubiquitous/concurrent-tmpGHU3ALSV.fasl":

    Permission denied

Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {1001878103}>
0: (SB-DEBUG::DEBUGGER-DISABLED-HOOK #<SB-INT:SIMPLE-FILE-ERROR "Error opening ~S" {100A6F0B43}> #<unused argument> :QUIT T)
1: (SB-DEBUG::RUN-HOOK *INVOKE-DEBUGGER-HOOK* #<SB-INT:SIMPLE-FILE-ERROR "Error opening ~S" {100A6F0B43}>)
2: (INVOKE-DEBUGGER #<SB-INT:SIMPLE-FILE-ERROR "Error opening ~S" {100A6F0B43}>)
3: (ERROR SB-INT:SIMPLE-FILE-ERROR :PATHNAME #P"/gnu/store/c4fj8gfdshgbfrm1jq3rdg55m24mkjh3-sbcl-ubiquitous-2.0.0-1.35eb7bd/lib/common-lisp/sbcl/ubiquitous/concurrent-tmpGHU3ALSV.fasl" :MESSAGE "Permission denied" :FORMAT-CONTROL "Error opening ~S" :FORMAT-ARGUMENTS (#P"/gn
u/store/c4fj8gfdshgbfrm1jq3rdg55m24mkjh3-sbcl-ubiquitous-2.0.0-1.35eb7bd/lib/common-lisp/sbcl/ubiquitous/concurrent-tmpGHU3ALSV.fasl"))
4: (SB-IMPL::FILE-PERROR #P"/gnu/store/c4fj8gfdshgbfrm1jq3rdg55m24mkjh3-sbcl-ubiquitous-2.0.0-1.35eb7bd/lib/common-lisp/sbcl/ubiquitous/concurrent-tmpGHU3ALSV.fasl" 13 "Error opening ~S" #P"/gnu/store/c4fj8gfdshgbfrm1jq3rdg55m24mkjh3-sbcl-ubiquitous-2.0.0-1.35eb7bd/lib/comm
on-lisp/sbcl/ubiquitous/concurrent-tmpGHU3ALSV.fasl")
---
 gnu/packages/lisp-xyz.scm | 53 +++++++++++++++++++++++++++++++++++++++
 1 file changed, 53 insertions(+)

diff --git a/gnu/packages/lisp-xyz.scm b/gnu/packages/lisp-xyz.scm
index 7cfd4b2a6b..7aea0c5b20 100644
--- a/gnu/packages/lisp-xyz.scm
+++ b/gnu/packages/lisp-xyz.scm
@@ -13975,3 +13975,56 @@ building Jupyter kernels, based on Maxima-Jupyter which was based on
 
 (define-public cl-common-lisp-jupyter
   (sbcl-package->cl-source-package sbcl-common-lisp-jupyter))
+
+(define-public sbcl-radiance
+  (let ((commit "5ffbe1f157edd17a13194495099efd81e052df85")
+        (revision "1"))
+    (package
+      (name "sbcl-radiance")
+      (version (git-version "2.1.2" revision commit))
+      (source
+       (origin
+         (method git-fetch)
+         (uri (git-reference
+               (url "https://github.com/Shirakumo/radiance")
+               (commit commit)))
+         (file-name (git-file-name name version))
+         (sha256
+          (base32 "0hbkcnmnlj1cqzbv18zmla2iwbl65kxilz9764hndf8x8as1539c"))))
+      (build-system asdf-build-system/sbcl)
+      (arguments
+       `(#:asd-files '("radiance-core.asd" "radiance.asd")
+         #:asd-systems '("radiance-core" "radiance")))
+      (inputs
+       `(("babel" ,sbcl-babel)
+         ("bordeaux-threads" ,sbcl-bordeaux-threads)
+         ("cl-ppcre" ,sbcl-cl-ppcre)
+         ("closer-mop" ,sbcl-closer-mop)
+         ("documentation-utils" ,sbcl-documentation-utils)
+         ("deploy" ,sbcl-deploy)
+         ("form-fiddle" ,sbcl-form-fiddle)
+         ("lambda-fiddle" ,sbcl-lambda-fiddle)
+         ("local-time" ,sbcl-local-time)
+         ("modularize-hooks" ,sbcl-modularize-hooks)
+         ("modularize-interfaces" ,sbcl-modularize-interfaces)
+         ("puri" ,sbcl-puri)
+         ("trivial-indent" ,sbcl-trivial-indent)
+         ("trivial-mimes" ,sbcl-trivial-mimes)
+         ("ubiquitous-concurrent" ,sbcl-ubiquitous)))
+      (home-page "https://shirakumo.github.io/radiance/")
+      (synopsis "Common Lisp web application environment")
+      (description
+       "Radiance is a web application environment, which is sort of like a web
+framework, but more general, more flexible.  It should let you write personal
+websites and generally deployable applications easily and in such a way that
+they can be used on practically any setup without having to undergo special
+adaptations.
+
+This package produces 2 systems: @code{RADIANCE} @code{RADIANCE-CORE}")
+      (license license:zlib))))
+
+(define-public ecl-radiance
+  (sbcl-package->ecl-package sbcl-radiance))
+
+(define-public cl-radiance
+  (sbcl-package->cl-source-package sbcl-radiance))
-- 
2.30.0

